src/uwtools/generic_scheduler/slurm.py

```diff
@@ -1,14 +1,44 @@
-from io import StringIO
 from .scheduler import Scheduler
+from datetime import datetime
+from pathlib import Path
 
 __all__ = ['Slurm']
 
-
 class Slurm(Scheduler):
-    pass
-def map_flags():
-    return
-def add_native_flag(flag):
-    pass
-def create_directive_list():
-    pass
\ No newline at end of file
+    _DIRECTIVE = '#SBATCH'
+
+    def __init__(self, scheduler: str, job_name: str, partition: list, qos: str, output: Path, error: Path, walltime: datetime, account: str, nodes: int, ntasks_per_node: int, ntasks: int, cpus_per_task: int, reservation: str, join: bool, native_flags: list, run_command: str):
+        super().__init__(scheduler, job_name, partition, qos, output, error, walltime, account, nodes, ntasks_per_node, ntasks, cpus_per_task, reservation, join, native_flags, run_command)
+        
+    def map_flags():
+        mappings = {'account': '-A', 'partition': '-p', 'wallclock': '-t', 'job_name': '-J', 'nodes': '-N', 'ntasks_per_node': '-n', 'output': '-o', 'error': '-e'}
```

Comment:
Here, I _think_ it could be useful to map in the other direction, so then you could map the value of the flag directly to the flag itself. 

You might also consider making it an object attribute in the __init__ method (instead of its own method here) so that you don't have to repeat the list of info, and all the values are stored in a convenient dict for retrieval elsewhere if needed.